client/state/login/reducer.js

```diff
@@ -25,6 +33,21 @@ export const isRequesting = createReducer( false, {
 	[ LOGIN_REQUEST_SUCCESS ]: () => false,
 } );
 
+export const redirectTo = createReducer( null, {
```

Comment:
The main reason for these changes was to better separate concerns in the state tree:

```json
login: {
  isRequesting: false,
  isRequestingTwoFactorAuth: false,
  magicLogin: {
    emailAddressFormInput: '',
    isFetchingEmail: false,
    requestAuthError: null,
    requestAuthSuccess: false,
    requestEmailError: null,
    requestEmailSuccess: false,
    currentView: null
  },
  redirectTo: 'https://haiku.blog/',
  rememberMe: true,
  requestError: null,
  requestNotice: null,
  requestSuccess: true,
  twoFactorAuth: {
    user_id: 12345,
    two_step_supported_auth_types: [
      'backup',
      'sms',
      'push'
    ],
    two_step_nonce_backup: 'obc8zkxbt1n9hk6zwbaijoa8ioi54y9h',
    two_step_nonce_sms: 'qs7qdntx5afn4dp65khc2juytzx7y32z',
    two_step_nonce_push: 'gv8n6nvws1ojrh6h4c1bacrwlk6g3pql',
    push_web_token: '1234567890:ju5ps6nl1h0ys5umz0tszphyk1kjs7x4tlsyr4f5nbo6sm2lpdzowi9l3x2j0kf4',
    two_step_notification_sent: 'push'
  },
  twoFactorAuthRequestError: null,
  twoFactorAuthPushPoll: {
    inProgress: true,
    success: false
  }
},
```

I don't think `redirectTo` and `rememberMe` should be stored under `login.twoFactorAuth` since they are more general login properties, and can be used in other flows such as magic login.

I also have the feeling that we would just be hiding that complexity/mess behind some selectors otherwise :).